app/scripts/components/map/index.js

```diff
@@ -140,14 +177,33 @@ const Map = React.createClass({
     this.draw.add(feature);
   },
 
+  splitLine: function (e) {
+    const { draw } = this;
+    const ids = draw.getFeatureIdsAt(e.point);
+    if (!ids.length) { return; }
+    const line = draw.get(ids[0]);
+    const cursorAt = point([e.lngLat.lng, e.lngLat.lat]);
+
+    // delete the existing line, and add two additional lines.
+    draw.delete(line.id);
+    const newIds = [];
+    newIds.push(draw.add(lineSlice(point(firstCoord(line)), cursorAt, line)));
+    newIds.push(draw.add(lineSlice(cursorAt, point(lastCoord(line)), line)));
+
+    this.splitMode({ featureIds: newIds });
+    const actions = newIds.map(id => createRedo(draw.get(id))).concat(createUndo(line));
```

Comment:
@dereklieu For the new lines (added in 190/192), can we also do `markAsEdited` so they appear blue?